contracts/lib/OrderValidator.sol

```diff
@@ -220,14 +222,64 @@ contract OrderValidator is Executor, ZoneInteraction {
                 }
             }
 
+            // Increment the filled numerator by the new numerator.
+            filledNumerator += numerator;
+
+            // Use assembly to ensure fractional amounts are below max uint120.
+            assembly {
+                // Check filledNumerator and denominator for uint120 overflow.
+                if or(
+                    gt(filledNumerator, MaxUint120),
+                    gt(denominator, MaxUint120)
+                ) {
+                    // Derive greatest common divisor using euclidean algorithm.
+                    function gcd(_a, _b) -> out {
+                        for {
+
+                        } gt(_b, 0) {
+
+                        } {
+                            let _c := _b
+                            _b := mod(_a, _c)
+                            _a := _c
+                        }
+                        out := _a
+                    }
+                    let scaleDown := gcd(
+                        numerator,
+                        gcd(filledNumerator, denominator)
+                    )
```

Comment:
Is it really necessary to do `gcd()` twice?  (costs gas and gets a possible worse value for `scaleDown`)
Because `numerator` isn't used in the rest of this function.
Except for `numerator := div(numerator, safeScaleDown)`, which isn't necessary then either.